File: abc128/b.py
n = int(input())

sps = []
for i in range(n):
    s, p = input().split()
    p = int(p)
    sps.append({
        "city": s,
        "score": p,
        "index": i
    })
# sps.sort()

cities = {}
for sp in sps:
    if not sp["city"] in cities:
        cities[sp["city"]] = []
    cities[sp["city"]].append(sp)

# [v.sort() for k, v in cities.items()]
rest_sorted_cities = {}
for k, v in cities.items():
    rest_sorted_cities[k] = sorted(v, key=lambda x:x["score"], reverse=True)

sorted_cities = sorted(rest_sorted_cities.items(), key=lambda x: x[0])

for k, city in sorted_cities:
    [print(rest["index"] + 1) for rest in city]

File: abc128/b.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <map>
using namespace std;

class Restaurant {
    public:
        string city;
        int score;
        
        Restaurant(string s, int p) {
            city = s;
            score = p;
        }
};

class City {
    string name;
    vector<Restaurant> restaurants;
    public:
        City(string name) {
            
        }

        void addRest(Restaurant rest) {
            restaurants.push_back(rest);
        }

        void sortRests() {
            sort(restaurants.begin(), restaurants.end(), &compare);
        }

        bool compare(Restaurant a, Restaurant b) {
            return a.score < b.score;
        }
};

int main()
{
    int n;
    cin >> n;

    map<string, City> cities;

    for (int i = 0; i < n; i++) {
        string s;
        int p;
        cin >> s >> p;
        Restaurant rest(s, p);

        if (cities.count(s) == 0) {
            cities[s] = City(s);
        }
        cities[s].addRest(rest);
    }
}
